### PR TITLE
Adds _ip to all segment calls; adds _userAgent to track call if event if

### PR DESCRIPTION
**Pull request by kyleaclark**

### CODE REVIEW

File: lib/mapper.js [hunk 1]
```diff
@@ -164,14 +168,16 @@ function createBaseTrackProperties(track, settings) {
...
     properties = prefix(track.event(), properties);
   }
-
-  return extend(
-    properties,
-    {
+  // Only apply userAgent property on `visited site` event
+  if (track.event() === 'visited site') {
+    properties._userAgent = userAgent || track.proxy('context.userAgent');
+  }
```
Comment (kyleaclark) [inline lib/mapper.js:174]: @paulgoleary Per the line code changes above, I want to confirm we (on the kissmetrics end) only parse `_userAgent` for the event `visited site`, and thus should only send the `_userAgent` string in the segment kissmetrics-integration when the event being tracked is `visted site`.  This update is per the following user story: https://www.pivotaltracker.com/n/projects/1392800/stories/109322764

cc: @percyhanna

Comment (percyhanna) [inline lib/mapper.js:174]: This implementation will require changes to `data-translator`. Currently we only parse the actual `User-Agent` header in the raw request. This is adding it as a separate property, which is not currently handled. @paulgoleary

Comment (paulgoleary) [inline lib/mapper.js:174]: Percy is correct. Seems there was a misunderstanding; _userAgent was not implemented, only _ip.
If this is a requirement it's also a straightforward add.

File: lib/mapper.js [hunk 2]
```diff
@@ -164,14 +168,16 @@ function createBaseTrackProperties(track, settings) {
   if (settings.prefixProperties) {
     properties = prefix(track.event(), properties);
   }
-
-  return extend(
-    properties,
-    {
+  // Only apply userAgent property on `visited site` event
+  if (track.event() === 'visited site') {
+    properties._userAgent = userAgent || track.proxy('context.userAgent');
```
Comment (percyhanna) [inline lib/mapper.js]: This has been implemented as `_ua`.

**Comment by NathanielWroblewski** [comment]
@paulgoleary @percyhanna will you guys please comment on anything that may need to be added here?  I'm going to take this branch over from kyle.  I've addressed the comments and tests are passing, but I'm less familiar with this feature request and what was implemented on the back-end for user agent.

File: lib/mapper.js [hunk 3]
```diff
@@ -164,14 +167,16 @@ function createBaseTrackProperties(track, settings) {
   if (settings.prefixProperties) {
     properties = prefix(track.event(), properties);
   }
-
-  return extend(
-    properties,
-    {
+  // Only apply userAgent property on `visited site` event
+  if (track.event() === 'visited site') {
```
Comment (percyhanna) [inline lib/mapper.js]: We should pass in `_ua` for all events, but we should not pass in empty/undefined user agents.

Comment (NathanielWroblewski) [inline lib/mapper.js]: What should be the value in those instances?

Should I change this then to something along the lines of:

```
properties._ua = track.userAgent() || track.proxy('context.userAgent') || 'DEFAULT_VALUE'
```

Comment (NathanielWroblewski) [inline lib/mapper.js]: I've added this to the PR.

**Comment by percyhanna** [comment]
Looks good to me. This PR can be merged, but really we need to open a PR against the original repo.

**Comment by percyhanna** [comment]
Has the `_ua` parameter been properly and fully tested through NDPP?

**Comment by NathanielWroblewski** [comment]
I haven't hear from @paulgoleary yet on the subject, but I'll reach out to him in the morning.

**Comment by NathanielWroblewski** [comment]
I'll merge for the time being, and when I hear from paul, I'll open against the original repo.

**Comment by NathanielWroblewski** [comment]
Good from paul and tested with ryan.  Merging.

**Pull request merged**